src/components/timers/Tabata.tsx

```diff
@@ -1,3 +1,304 @@
+            clearInterval(intervalRef.current);
+            intervalRef.current = null;
+        }
+    };
+
+    const initialCountdown = () => {
+        setStatus(STATUS.INITIAL);
+    };
+
+    //this effect takes care of reps when seconds for both work and rest are complete
+    useEffect(() => {
+        if (status === STATUS.STARTED && secondsRemainingRest === 0 && secondsRemainingWork === 0 && repRemaining > 0) {
+            const totalSecondsWork = convertToSeconds(timeMinInputWork, timeSecInputWork);
+            const totalSecondsRest = convertToSeconds(timeMinInputRest, timeSecInputRest);
+
+            setRepRemaining(prevRep => prevRep - 1);
+
+            if (repRemaining - 1 <= 0) {
+                setStatus(STATUS.STOPPED);
+            } else {
+                setSecondsRemainingWork(totalSecondsWork);
+                setSecondsRemainingRest(totalSecondsRest);
+                setSecondsRemainingTotal(totalSecondsWork + totalSecondsRest);
+            }
+        }
+    }, [secondsRemainingWork, secondsRemainingRest, status, repRemaining, timeMinInputWork, timeMinInputRest, timeSecInputRest, timeSecInputWork]);
+
+    //this effect takes care of the seconds
+    useEffect(() => {
+        if (status === STATUS.STARTED) {
+            //this decrements the seconds immideately without waiting for the first second – only implemented for the very first second when reps are still intact
+            if (secondsRemainingWork === totalSecondsWork && repRemaining === Number(repInput)) {
+                setSecondsRemainingWork(prevWork => {
+                    if (prevWork > 1) return prevWork - 1;
+                    return prevWork; //this is added to fix a TypeScript error to ensure we never return undefined
+                });
+            } else if (secondsRemainingWork > 0) {
+                intervalRef.current = setInterval(() => {
+                    setSecondsRemainingWork(prevWork => {
+                        if (prevWork > 1) {
+                            return prevWork - 1;
+                        } else {
+                            return 0;
+                        }
+                    });
+                }, 1000);
+            } else if (secondsRemainingWork === 0 && secondsRemainingRest > 0) {
+                intervalRef.current = setInterval(() => {
+                    setSecondsRemainingRest(prevRest => {
+                        if (prevRest > 0) {
+                            return prevRest - 1;
+                        } else {
+                            return 0;
+                        }
+                    });
+                }, 1000);
+            }
+        }
+
+        return () => {
+            if (intervalRef.current !== null) {
+                clearInterval(intervalRef.current);
+                intervalRef.current = null;
+            }
+        };
+    }, [status, secondsRemainingWork, secondsRemainingRest, repInput, repRemaining, totalSecondsWork]);
+
+    return (
```

Comment:
Great job displaying all relevant information (there's a lot!) for this Tabata timer. One thing you could consider - change the background color when in rest mode. The text changing from `work left` to `rest left` is a bit subtle, might not be noticeable by users who are busy working out.